src/peer_manager.rs

```diff
@@ -442,25 +448,15 @@ impl PeerManager {
 
     /// Upgrades a `Bootstrapper` to a `Client` or `JoiningNode`.
     pub fn handle_bootstrap_request(&mut self, pub_id: &PublicId) {
-        let self_debug = format!("{:?}", self);
-        match self.peers.get_mut(pub_id) {
-            Some(peer @ &mut Peer {
-                          state: PeerState::Bootstrapper(CrustUser::Client), ..
-                      }) => {
-                debug!("{} Accepted Client {}.", self_debug, pub_id);
-                peer.state = PeerState::Client;
-            }
-            Some(peer @ &mut Peer { state: PeerState::Bootstrapper(CrustUser::Node), .. }) => {
-                debug!("{} Accepted JoiningNode {}.", self_debug, pub_id);
-                peer.state = PeerState::JoiningNode;
-            }
-            _ => {
-                log_or_panic!(LogLevel::Error,
-                              "{} {:?} is not a bootstrapper.",
-                              self_debug,
-                              pub_id);
-            }
-        }
+        let _ = self.peers
```

Comment:
This one is still lacking - although the else has been added below with log or panic, it does not cover the case where we don't find `pub_id` in `self.peers` at all - previously we panicked on that, not anymore. So needs done. Better to retain the previous format and just add the `ip` part to it ?